app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -3,64 +3,144 @@ package com.bignerdranch.android.geoquiz
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.view.View
-import android.widget.Toast
+import android.widget.*
+import androidx.lifecycle.ViewModelProviders
 import com.bignerdranch.android.geoquiz.databinding.ActivityMainBinding
 
+private const val KEY_INDEX = "index"
+private const val KEY_SCORE = "score"
+private const val KEY_TOTALANSWERS = "total_answers"
+
 class MainActivity : AppCompatActivity() {
+
     private lateinit var binding: ActivityMainBinding
-    private var currentIndex = 0
+    private val quizViewModel: QuizViewModel by lazy {
+        ViewModelProviders.of(this).get(QuizViewModel::class.java)
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setupViewBinding()
+        getSavedInstanceState(savedInstanceState)
         setupClickListener()
+        checkQuestion()
         updateQuestion()
     }
 
+    override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        savedInstanceState.putInt(KEY_INDEX, quizViewModel.currentIndex)
+        savedInstanceState.putInt(KEY_SCORE, quizViewModel.score)
+        savedInstanceState.putInt(KEY_TOTALANSWERS, quizViewModel.answersCounter)
+    }
+
+    private fun setupViewBinding() {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+    }
+
+    private fun getSavedInstanceState(savedInstanceState: Bundle?) {
+        val currentIndex = savedInstanceState?.getInt(KEY_INDEX, 0) ?: 0
```

Comment:
not now because you have not worked with live data, but in the future, you can use https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate